packages/core/src/actor/instance.ts

```diff
@@ -311,28 +312,38 @@ export class ActorInstance<
 
 		logger().info("actor ready");
 		this.#ready = true;
+
+		this.#scheduleLivenessCheck();
 	}
 
-	async scheduleEvent(
+	async #scheduleInternalEvent(
 		timestamp: number,
-		fn: string,
-		args: unknown[],
-	): Promise<void> {
+		event: {
+			fn: string | ScheduledInternalEvent;
```

Comment:
same issue as last review. you're mixing together internal events and user action events, such that internal even names will clobber user event names.

the event type needs to be an ADT

something like:

```
type EventType = { action: { name: string, args: unknown[] } } | { checkConnLiveness: {} };
```